File: Algorithm-CPP/16_Anagram.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    string a, b;

    cin >> a;
    cin >> b;

    if (a.length() != b.length()) {
        cout << "NO" << endl;
        return 0;
    }

    sort(a.begin(), a.end());
    sort(b.begin(), b.end());

    for(int i = 0; a[i] != '\0'; i++) {
        if (a[i] != b[i]) {
            cout << "NO" << endl;
            return 0;
        }
    }

    cout << "YES" << endl;

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/16_Anagram.swift
//
//  16_Anagram.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/07/30.
//

import Foundation

func solution16() {
    let a = readLine()!
    let b = readLine()!

    if a.count != b.count {
        print("NO")
        return
    }

    let sortedA = a.sorted()
    let sortedB = b.sorted()
    for i in 0..<sortedA.count {
        if sortedA[i] != sortedB[i] {
            print("NO")
            return
        }
    }

    print("YES")
}
